sdk/main/include/ECDSAsecp256k1PrivateKey.h

```diff
@@ -43,14 +45,10 @@ class ECDSAsecp256k1PrivateKey : public PrivateKey
    */
   ECDSAsecp256k1PrivateKey() = delete;
 
-  /**
-   * Destructor
-   */
-  ~ECDSAsecp256k1PrivateKey() override;
-
   /**
    * Copy constructor and copy assignment operator can throw std::runtime_error if OpenSSL serialization fails.
    */
+  ~ECDSAsecp256k1PrivateKey() override = default;
```

Comment:
Do we actually need to declare that we are using the default destructor on all of these classes? Won't it use the default one anyway if you just say nothing?

Reply:
It's always been my impression that it's good practice to, if you have to define one of copy constructor, copy assignment operator, move constructor, move assignment operator, or destructor, then you should define them all, even if they're `default`. With that said, removing this line doesn't have any effect and all tests/examples still pass with no leaks (even tried some stuff that I thought could cause some undefined behavior but it worked fine) so I'm fine removing it if you would like

Reply:
I'm going to merge this as is for now and can come back to it at some point. I was planning on reviewing/updating the implementation of these classes at some point anyways for a couple reasons but primarily to see if there's a way to get rid of these functions. Ideally none of these functions would be needed, but since we want to allow copying and moving of keys (I would think) and this class has `std::unique_ptr` members, they need to be defined.